rst2pdf/extensions/preprocess_r2p.py

```diff
@@ -108,12 +108,20 @@
 
 from rst2pdf.log import log
 
-class DummyFile(str):
+
+class DummyFile(object):
     ''' We could use stringio, but that's really overkill for what
         we need here.
     '''
+    def __init__(self, value):
+        self._str = value
+
     def read(self):
-        return self
+        return self._str
+
+    def strip(self):
```

Comment:
File objects don't usually have a `strip` method - that's a method on `str` etc.

I can see you've added  a `read` below - this needs to be moved further up. I'll do this now and add the commit.